syntax = "proto3";
package instance_host;
option go_package = "github.com/spielbrett/spielbrett-api/gen/go/proto";

message CreateInstanceRequest {
  string instance_type = 1;
}

message CreateInstanceResponse {
  string instance_id = 1;
}

message PerformActionRequest {
  string instance_id = 1;
  string user_id = 2;
  string action_name = 3;
  string payload = 4;
}

message PerformActionResponse {
  string payload = 1;
}

service InstanceHostService {
  rpc CreateInstance(CreateInstanceRequest) returns (CreateInstanceResponse) {}
  rpc PerformAction(PerformActionRequest) returns (PerformActionResponse) {}
}
